tests/models/test_xcom_arg.py

```diff
@@ -0,0 +1,146 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from datetime import datetime, timedelta
+
+from airflow import DAG
+from airflow.models.xcom_arg import XComArg
+from airflow.operators.bash import BashOperator
+from airflow.operators.python import PythonOperator
+
+DEFAULT_ARGS = {
+    "owner": "test",
+    "depends_on_past": True,
+    "start_date": datetime.today(),
+    "retries": 1,
+    "retry_delay": timedelta(minutes=1),
+}
+
+VALUE = 42
+
+
+def assert_is_value(num: int):
+    assert num == VALUE
+
+
+def build_python_op():
+    def f(task_id):
+        return f"OP:{task_id}"
+
+    with DAG(dag_id="test_xcom_dag", default_args=DEFAULT_ARGS):
+        operator = PythonOperator(
+            python_callable=f,
+            task_id="test_xcom_op",
+            do_xcom_push=True,
+        )
+        return operator
+
+
+class TestXComArgBuild:
+    def test_xcom_ctor(self):
+        python_op = build_python_op()
+        actual = XComArg(python_op, "test_key")
+        assert actual
+        assert actual.operator == python_op
+        assert actual.key == "test_key"
+        assert actual == actual  # pylint: disable=comparison-with-itself
```

Comment:
Why this check?

Reply:
To assert the __eq__ method.
Added this as a comment :)